pysand/erosion.py

```diff
@@ -394,6 +392,36 @@ def choke_gallery(v_m, rho_m, mu_m, GF, D, d_p, R_c, gap, H, material='cr_37_tun
 
     return E_rel
 
+def nozzlevalve_wall(v_m, d_p, GF, At, material='duplex'):
+    """
+    Particle valve wall erosion for non-slam nozzle type check-valve. Based on DNVGL CFD-study of Johan Sverdrup Phase 1 check-valves (13.01.2020)
+    Report No: 2019-1237 Rev.1, Document No: 547341
+    :param v_m: Internal valve mix velocity [m/s]. Use mixture velocity through minimum flow area of the valve.
+    :param GF: Geometry factor [-]
+    :param At: Target area [m²]. Set to minimum flow area of the valve. 
+    :param d_p: Particle diameter [mm]
+    :param material: Material exposed to erosion, default = 'duplex'. For others, run: materials()
+    :return: Relative erosion rate [mm/ton]
+    """
+    # Input validation
+    kwargs = {'v_m': v_m, 'd_p': d_p, 'GF': GF, 'At': At}
+    if validate_inputs(**kwargs):
+        return np.nan
+    
+    # Stricter particle size validation
+    if (d_p > 0.6):
```

Comment:
This is input validation which maybe should be done in the validation function instead.